src/channel/rpc_0_7.ts

```diff
@@ -168,6 +169,22 @@ export class RpcChannel {
     return this.fetchEndpoint('starknet_getBlockWithTxs', { block_id });
   }
 
+  public async getL1MessageHash(txHash: BigNumberish) {
+    const txData = (await this.getTransactionByHash(txHash)) as any;
+    const { calldata, contract_address, entry_point_selector, nonce, version, type } = txData;
+    const hashedCallData = computeHashOnElements(calldata);
+
+    return computeHashOnElements([
+      useEncoded(type),
+      version,
+      contract_address,
+      entry_point_selector,
+      hashedCallData,
+      '0x0',
+      nonce,
+    ]);
+  }
+
```

Comment:
Are you sure this is algorithm?
With this transaction Hash : `https://sepolia.voyager.online/tx/0x28dfc05eb4f261b37ddad451ff22f1d08d4e3c24dc646af0ec69fa20e096819#messages`
The result has to be `0x55b3f8b6e607fffd9b4d843dfe8f9b5c05822cd94fcad8797deb01d77805532a`